message/pool/message.go

```diff
@@ -560,3 +560,26 @@ func (r *Message) SetupPut(path string, token message.Token, contentFormat messa
 func (r *Message) SetupDelete(path string, token message.Token, opts ...message.Option) error {
 	return r.setupCommon(codes.DELETE, path, token, opts...)
 }
+
+func (r *Message) Clone(p *Pool) (*Message, error) {
+	msg := p.AcquireMessage(r.Context())
+	msg.SetCode(r.Code())
+	msg.SetToken(r.Token())
+	msg.ResetOptionsTo(r.Options())
+	msg.SetType(r.Type())
+	msg.SetMessageID(r.MessageID())
+	if r.Body() != nil {
+		defer func() {
+			_, _ = r.Body().Seek(0, io.SeekStart)
```

Comment:
Can `Seek` fail here? It might corrupt the data in that case, so at least logging it would be nice.

Reply:
Added